common/lib/client/auth.js

```diff
@@ -38,27 +38,25 @@ var Auth = (function() {
 		return JSON.stringify(c14nCapability);
 	}
 
+	function containsAuthInfo(options) {
+		var authMethods = ['authUrl', 'authCallback', 'token', 'tokenDetails', 'key'];
+		return (Utils.arrAny(authMethods, function(method) {
+			return (method in options);
+		}));
+	}
+
 	function Auth(client, options) {
 		this.client = client;
 		this.tokenParams = options.defaultTokenParams || {};
 
-		/* RSA7a4: if options.clientId is provided and is not
-		 * null, it overrides defaultTokenParams.clientId */
-		if(options.clientId) {
```

Comment:
It still will, it's just now done by the `_setClientId` stuff which is called by the `_save` methods. DRYer this way, don't need to have RSA7a4 logic in three places. (note that the `init_conflicting_clientids` test still passes)